R/sample_pseudoabs.R

```diff
@@ -41,11 +41,17 @@
 
 sample_pseudoabs <- function (data, raster, n, coords = NULL,
                               method="random", class_label = "pseudoabs",
-                              return_pres=TRUE)
+                              return_pres = TRUE)
 {
   return_sf <- FALSE # flag whether we need to return an sf object
-  if (inherits(data,"sf")){
-    data <- data %>% dplyr::bind_cols(sf::st_coordinates(data))
+  if (inherits(data,"sf")) {
+    bind_col <- TRUE
```

Comment:
Hmm, I am not sure this solves the problem. If you had X and Y that are different from the ones that would come from the geometry, the current code would go ahead with binding, and then generate the problem that you have already observed. So, I think what we need to do is check if we already have those names (X and Y), and IF we do, check that they are informative. If not, we can either throw an error, or rename the current X and Y, and replace them with new X and Y from geometry. If we go for the latter, there should be a message informing the user that their variables got relabelled.
To avoid unwanted side effects, it would be ideal if a test for this behaviour is added to `/test/testthat/test_sample_pseudoabs`. Thanks!